drivers/mtd/smart.c

```diff
@@ -331,6 +331,10 @@ static bool inline smart_is_free(FAR struct smart_struct_s *dev, uint16_t sector
 static void smart_save_meta(FAR struct smart_struct_s *dev)
 {
   /* save sMap at block 1 */
+  if (dev->mapphyssector == SMART_SECTOR_INVALID) {
```

Comment:
The mapphyssector should be valid sector, after smart_scan. In which case it is invalid?